src/item_factory.cpp

```diff
@@ -703,6 +703,12 @@ void Item_factory::load( islot_ammo &slot, JsonObject &jo )
     slot.recoil = jo.get_int( "recoil", 0 );
     slot.def_charges = jo.get_long( "count" );
     slot.ammo_effects = jo.get_tags( "effects" );
+
+    if( jo.has_int( "loudness" ) ) {
```

Comment:
You could use `JsonObject::read` to avoid some repetition:
```C++
if( !jo.read( "loudness", slot.loudness ) ) {
    slot.loudness = default_value...
}
```